vignettes/derive_single_dose_fillin.Rmd

```diff
@@ -0,0 +1,118 @@
+may be required. This programming strategy allows the maximum possible number 
+of single dose records to be derived.
+The example R script requires the date of datacut DCUTDT to be specified correctly
+All available trial treatments are included, allowing multiple different 
+last dose variables to be created in ad_adae.R if required.
+
+
+Load required packages ----
+```{r}
+library(admiral)
+library(admiral.test) # Contains example datasets from the CDISC pilot project
+library(dplyr)
+library(lubridate)
+library(tidyr)
+
+data("admiral_dm")
+data("admiral_ex")
+
+dm <- admiral_dm
+ex <- admiral_ex
+
+ex <- convert_blanks_to_na(ex)
+dm <- convert_blanks_to_na(dm)
+```
+Derive date of death and set up data cut date
+to fill in missing treatment end date EXENDT 
+and missing treatment end datetime EXENDTM
+Alternative approach instead of death date could use last known alive date ADSL LSTALVDT
+```{r}
+dm_death <- dm %>%
+  select(STUDYID, USUBJID, DTHDTC) %>%
+  derive_vars_dtm(
+    dtc = DTHDTC,
+    flag_imputation = "none",
+    new_vars_prefix = "DTH"
+  ) %>% 
+ derive_vars_dtm_to_dt(vars(DTHDTM)) %>%
+# Set up datacut as character snapshot date to use in EXENDT and EXENDTM imputations
+  mutate(DCUTDT = as.Date("2015-03-06"),
+         DCUTDTM = convert_date_to_dtm(DCUTDT))
+
+```
+Select valid dose records, non-missing EXSTDTC and EXDOSE
+ Set up missing EXDOSFRQ as QD daily dosing regime
+```{r}
+ex_mod <- ex %>%
+ filter(!is.na(EXSTDTC) & !is.na(EXDOSE)) %>%
+  left_join (dm_death, by=c("STUDYID","USUBJID")) %>% 
+  mutate (EXDOSFRQ = if_else (is.na(EXDOSFRQ), "QD", EXDOSFRQ) 
+     )%>%
+  derive_vars_dtm(
+    dtc = EXSTDTC,
+    new_vars_prefix = "EXST",
+    date_imputation = "first",
+    time_imputation = "first",
+    flag_imputation = "none",
+  ) %>%
+ derive_vars_dtm_to_dt(vars(EXSTDTM)) %>%
+  derive_vars_dt(
+    dtc = EXENDTC,
+    new_vars_prefix = "EXEN",
+    # Maximum imputed treatment end date must not be not greater than date of death
+    # or after the datacut date
+    max_dates = vars(DTHDT,DCUTDT),
+    date_imputation = "last"
+  ) %>% 
+  derive_vars_dtm(
+    dtc = EXENDTC,
+    new_vars_prefix = "EXEN",
+    # Maximum imputed treatment end date must not be not greater than date of death
+    # or after the datacut date
+    max_dates = vars(DTHDTM,DCUTDTM),
+    date_imputation = "last",
+    time_imputation = "last",
+    flag_imputation = "none")  %>% 
+  mutate(
+    EXENDT = if_else (is.na(EXENDT) & is.na(DTHDT), DCUTDT, EXENDT),
+    EXENDT = if_else (is.na(EXENDT) & !is.na(DTHDT), DTHDT, EXENDT),
+    EXENDTM = if_else (is.na(EXENDTM) & is.na(DTHDTM), DCUTDTM, EXENDTM), 
+    EXENDTM = if_else (is.na(EXENDTM) & !is.na(DTHDTM), DTHDTM, EXENDTM) 
+   )%>%
+distinct(STUDYID,USUBJID,EXTRT,EXDOSE,EXDOSFRQ,DCUTDT,DTHDT,EXSTDT,EXSTDTM,EXENDT,EXENDTM,EXSTDTC,EXENDTC)
+
+ex_single<-
+  create_single_dose_dataset(
+    ex_mod,
+    dose_freq = EXDOSFRQ,
```

Comment:
I would not specify `dose_freq`, `lookup_table`, and `lookup_column` because the values are the same as the default values. If `dose_freq_lookup` is renamed in admiral and the default value is updated, the code would fail.

Reply:
I will make the changes